faucet/components/Faucet.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react'
+import { isAddress } from 'web3-utils'
+import type { Networks } from '@hoprnet/hopr-ethereum/scripts/addresses'
+
+async function mint(
+  network: Networks,
+  address: string
+): Promise<{
+  success: boolean
+  transactionHash?: string
+  message?: string
+}> {
+  const response = await fetch(`/api/mint?network=${network}&address=${address}`)
+
+  if (response.status === 200) {
+    const { transactionHash } = await response.json()
+    return {
+      success: true,
+      transactionHash,
+    }
+  } else {
+    return {
+      success: false,
+      message: response.statusText,
```

Comment:
I feel this might be a bit too verbose for what we are doing. We are doing in 14 lines the following:
```
const thrower = (err:string) => { throw Error(err); } // this can be in another file e.g. utils
  return await fetch(`/api/mint?network=${network}&address=${address}`)
    .then(res => res.status === 200 ? res.json() : thrower('Wrong status code') )
    .then(txReceipt => ({ success: true, transactionHash: txReceipt.transactionHash })
    .catch(err => { success: false, message: err })
```